src/main/java/gregtech/api/recipes/RecipeBuilder.java

```diff
@@ -390,14 +388,28 @@ public R append(Recipe recipe, int multiplier, boolean multiplyDuration) {
         // Build the new Recipe with multiplied components
         this.inputsIngredients(newRecipeInputs);
         this.fluidInputs(newFluidInputs);
-        this.outputs(outputItems);
+        if (trimOutputs && !outputItems.isEmpty()) {
+            this.outputs(outputItems.stream()
+                    .filter(is ->
+                            ItemStackHashStrategy.comparingAllButCount()
+                                    .equals(is, recipe.getOutputs().get(0)))
+                    .collect(Collectors.toList()));
+        } else {
+            this.outputs(outputItems);
+        }
         this.fluidOutputs(outputFluids);
 
         this.EUt(multiplyDuration ? recipe.getEUt() : this.EUt + recipe.getEUt() * multiplier);
         this.duration(multiplyDuration ? this.duration + recipe.getDuration() * multiplier : recipe.getDuration());
         this.parallel += multiplier;
 
-        chancedOutputsMultiply(recipe, multiplier);
+        if (!trimOutputs) {
+            chancedOutputsMultiply(recipe, multiplier);
+        } else if (this.outputs.size() == 0) {
+            ItemStack firstChancedOutput = recipe.getChancedOutputs().get(0).getItemStack().copy();
+            firstChancedOutput.setCount(firstChancedOutput.getCount() * multiplier);
+            this.outputs(firstChancedOutput);
```

Comment:
If we are trimming the output and there are no non-chanced outputs, the result of the trim should remain a chanced output instead of a guaranteed output. It looks like in this case where there are no outputs, it is guaranteeing an output.

You could instead do the `chancedOutputsMultiply` onto a dummy recipe, and then if we are trimming the outputs and there are no non-chanced outputs, only add the first resulting chanced entry from the `chancedOutputsMultiply`, and add the whole thing if we are not trimming.